File: Frontend/students-employment-records-site/src/domain/students/enums/gender.ts
import { enumToArrayNumber } from "@/tools/enums/enumUtils"
import { NeverUnreachable } from "@/tools/neverUreachable"

export enum Gender {
    Male = 1,
    Female = 2,
}

export namespace Gender {
    export function getAll(): Gender[] {
        return enumToArrayNumber<Gender>(Gender)
    }

    export function getDisplayText(gender: Gender): string {
        switch (gender) {
            case Gender.Male:
                return "Мужской"
            case Gender.Female:
                return "Женский"
            default:
                throw new NeverUnreachable(gender)
        }
    }
}

File: Backend/SER.Domain/Students/Enums/Gender.cs
namespace SER.Domain.Students.Enums;
public enum Gender
{
	Male = 1,
	Female = 2
}

public static class GenderExtensions
{
	public static String DisplayName(this Gender gender)
	{
		return gender switch
		{
			Gender.Male => "М",
			Gender.Female => "Ж",
			_ => throw new Exception()
		};
	}
}
